app/main.py

```diff
@@ -1,6 +1,22 @@
+from functools import wraps
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> int:
+        cache_key = (args, frozenset(kwargs.items()))
+
+        if cache_key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[cache_key]
+
+        print("Calculating new result")
+        result = func(*args, **kwargs)
+        cache_dict[cache_key] = result
```

Comment:
you can make this in `if not` statement, and unify your return by some result variable, so you will remove one of returns